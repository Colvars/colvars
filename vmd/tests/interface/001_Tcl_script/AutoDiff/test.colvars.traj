# step         rmsd                  fa_rmsd                rmsd-all               E_harmonic1           x0_rmsd               W_harmonic1           
           0    7.02166693715340e-17 -7.02166693715340e-13   0.00000000000000e+00   2.46519032881566e-29  0.00000000000000e+00  0.00000000000000e+00 
           1    3.16374866030855e-03 -3.16374866030855e+01   3.06562197951411e-03   5.00465279280208e-02  0.00000000000000e+00  0.00000000000000e+00 
           2    6.43042836588681e-03 -6.43042836588681e+01   6.20782631005234e-03   2.06752044844009e-01  0.00000000000000e+00  0.00000000000000e+00 
           3    9.79623008237720e-03 -9.79623008237720e+01   9.43628625319812e-03   4.79830619134360e-01  0.00000000000000e+00  0.00000000000000e+00 
           4    1.32402365614075e-02 -1.32402365614075e+02   1.27755653235612e-02   8.76519321010162e-01  0.00000000000000e+00  0.00000000000000e+00 
           5    1.67526858867062e-02 -1.67526858867062e+02   1.62397875984790e-02   1.40326242209322e+00  0.00000000000000e+00  0.00000000000000e+00 
           6    2.05399880743223e-02 -2.05399880743223e+02   1.98096952490138e-02   2.10945555046652e+00  0.00000000000000e+00  0.00000000000000e+00 
           7    2.46472075956544e-02 -2.46472075956544e+02   2.34704400098299e-02   3.03742421131643e+00  0.00000000000000e+00  0.00000000000000e+00 
           8    2.88885276374936e-02 -2.88885276374936e+02   2.71557732913004e-02   4.17273514531116e+00  0.00000000000000e+00  0.00000000000000e+00 
           9    3.34520835953681e-02 -3.34520835953681e+02   3.08846350501962e-02   5.59520948435747e+00  0.00000000000000e+00  0.00000000000000e+00 
          10    3.82640246532821e-02 -3.82640246532821e+02   3.46452320983256e-02   7.32067791333490e+00  0.00000000000000e+00  0.00000000000000e+00 
