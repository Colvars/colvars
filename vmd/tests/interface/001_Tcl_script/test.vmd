# -*- tcl -*-
mol new ../Common/da.psf type psf
mol addfile ../Common/da.pdb type pdb waitfor all
# this is the file from NAMD's 3rd reg test: it is XYZ format to gain more precision
mol addfile ../Common/da.test-traj.xyz type xyz waitfor all
# load again the PSF file to restore correct masses (XYZ parser temporarily screws them up)
mol addfile ../Common/da.psf type psf waitfor all

cv molid top
cv configfile test.in

cv delete
cv molid top
cv configfile test.in

# add the RMSD of all heavy atoms
set noh [atomselect top "not hydrogen"]
cv config "
colvar {
    name rmsd-all
 
    rmsd {
        atoms {
            atomNumbers { [$noh get serial] }
        }
        refpositionsfile ../Common/da.pdb
    }
}
"

set out [open "test.colvars.traj" "w"]
puts -nonewline ${out} [cv printframelabels]
for { set frame 0 } { ${frame} < [molinfo top get numframes] } { incr frame } {
    cv frame ${frame}
    cv update
    puts -nonewline ${out} [cv printframe]
}
close ${out}

quit
