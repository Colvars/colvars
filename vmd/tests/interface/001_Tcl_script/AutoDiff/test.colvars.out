colvars: ----------------------------------------------------------------------
colvars: Please cite Fiorin et al, Mol Phys 2013:
colvars:   https://doi.org/10.1080/00268976.2013.813594
colvars: as well as all other papers listed below for individual features used.
colvars: This version was built with the C++11 standard or higher.
colvars: Redefining the Tcl "cv" command to the new script interface.
colvars: ----------------------------------------------------------------------
colvars: Reading new configuration from file "test.in":
colvars: # units = "" [default]
colvars: # smp = on [default]
colvars: # colvarsTrajFrequency = 1
colvars: # colvarsRestartFrequency = 0
colvars: # scriptedColvarForces = off [default]
colvars: # scriptingAfterBiases = off [default]
colvars: # sourceTclFile = "" [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = "rmsd"
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "" [default]
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerToOrigin = off [default]
colvars:       # centerToReference = off [default]
colvars:       # rotateToReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID = { BH }
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 10 atoms requested: total mass = 120.11, total charge = 0.53.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/da.pdb"
colvars:     # refPositionsCol = "" [default]
colvars:     Enabling "centerToReference" and "rotateToReference", to minimize RMSD before calculating it as a variable: if this is not the desired behavior, disable them explicitly within the "atoms" block.
colvars:     This is a standard minimum RMSD, derivatives of the optimal rotation will not be computed as they cancel out in the gradients.
colvars:   All components initialized.
colvars:   # timeStepFactor = 1 [default]
colvars:   # width = 0.01
colvars:   # lowerBoundary = 0 [default]
colvars:   # upperBoundary = 0.1
colvars:   # hardLowerBoundary = on [default]
colvars:   # hardUpperBoundary = off [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputTotalForce = off [default]
colvars:   # outputAppliedForce = on
colvars:   # subtractAppliedForce = off [default]
colvars:   # runAve = off [default]
colvars:   # corrFunc = off [default]
colvars: ----------------------------------------------------------------------
colvars: Collective variables initialized, 1 in total.
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new "harmonic" instance.
colvars:   # name = "harmonic1" [default]
colvars:   # colvars = { rmsd }
colvars:   # stepZeroData = off [default]
colvars:   # outputEnergy = on
colvars:   # outputFreq = 0 [default]
colvars:   # timeStepFactor = 1 [default]
colvars:   # writeTISamples = off [default]
colvars:   # writeTIPMF = off [default]
colvars:   # centers = { 0 }
colvars:   # targetCenters = { 0.2 }
colvars:   # targetNumSteps = 20
colvars:   # targetNumStages = 0 [default]
colvars:   # outputAccumulatedWork = on
colvars:   # outputCenters = on
colvars:   # forceConstant = 1 [default]
colvars:   # decoupling = off [default]
colvars:   # targetForceConstant = -1 [default]
colvars:   The force constant for colvar "rmsd" will be rescaled to 10000 according to the specified width (0.01).
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new "harmonicwalls" instance.
colvars:   # name = "wall_rmsd"
colvars:   # colvars = { rmsd }
colvars:   # stepZeroData = off [default]
colvars:   # outputEnergy = off [default]
colvars:   # outputFreq = 0 [default]
colvars:   # timeStepFactor = 1 [default]
colvars:   # writeTISamples = off [default]
colvars:   # writeTIPMF = off [default]
colvars:   # forceConstant = 1 [default]
colvars:   # decoupling = off [default]
colvars:   # targetForceConstant = -1 [default]
colvars:   # lowerWalls = { 0 } [default]
colvars:   Lower walls were not provided.
colvars:   # upperWalls = { 0.1 }
colvars:   # upperWallConstant = 10
colvars:   The upper wall force constant for colvar "rmsd" will be rescaled to 100000 according to the specified width (0.01).
colvars: ----------------------------------------------------------------------
colvars: Collective variables biases initialized, 2 in total.
colvars: ----------------------------------------------------------------------
colvars: Collective variables module (re)initialized.
colvars: ----------------------------------------------------------------------
colvars: Re-initialized atom group for variable "rmsd":0/0. 10 atoms: total mass = 120.11, total charge = 0.53.
colvars: Resetting the Collective Variables module.
colvars: ----------------------------------------------------------------------
colvars: Please cite Fiorin et al, Mol Phys 2013:
colvars:   https://doi.org/10.1080/00268976.2013.813594
colvars: as well as all other papers listed below for individual features used.
colvars: This version was built with the C++11 standard or higher.
colvars: Redefining the Tcl "cv" command to the new script interface.
colvars: ----------------------------------------------------------------------
colvars: Reading new configuration from file "test.in":
colvars: # units = "" [default]
colvars: # smp = on [default]
colvars: # colvarsTrajFrequency = 1
colvars: # colvarsRestartFrequency = 0
colvars: # scriptedColvarForces = off [default]
colvars: # scriptingAfterBiases = off [default]
colvars: # sourceTclFile = "" [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = "rmsd"
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "" [default]
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerToOrigin = off [default]
colvars:       # centerToReference = off [default]
colvars:       # rotateToReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID = { BH }
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 10 atoms requested: total mass = 120.11, total charge = 0.53.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/da.pdb"
colvars:     # refPositionsCol = "" [default]
colvars:     Enabling "centerToReference" and "rotateToReference", to minimize RMSD before calculating it as a variable: if this is not the desired behavior, disable them explicitly within the "atoms" block.
colvars:     This is a standard minimum RMSD, derivatives of the optimal rotation will not be computed as they cancel out in the gradients.
colvars:   All components initialized.
colvars:   # timeStepFactor = 1 [default]
colvars:   # width = 0.01
colvars:   # lowerBoundary = 0 [default]
colvars:   # upperBoundary = 0.1
colvars:   # hardLowerBoundary = on [default]
colvars:   # hardUpperBoundary = off [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputTotalForce = off [default]
colvars:   # outputAppliedForce = on
colvars:   # subtractAppliedForce = off [default]
colvars:   # runAve = off [default]
colvars:   # corrFunc = off [default]
colvars: ----------------------------------------------------------------------
colvars: Collective variables initialized, 1 in total.
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new "harmonic" instance.
colvars:   # name = "harmonic1" [default]
colvars:   # colvars = { rmsd }
colvars:   # stepZeroData = off [default]
colvars:   # outputEnergy = on
colvars:   # outputFreq = 0 [default]
colvars:   # timeStepFactor = 1 [default]
colvars:   # writeTISamples = off [default]
colvars:   # writeTIPMF = off [default]
colvars:   # centers = { 0 }
colvars:   # targetCenters = { 0.2 }
colvars:   # targetNumSteps = 20
colvars:   # targetNumStages = 0 [default]
colvars:   # outputAccumulatedWork = on
colvars:   # outputCenters = on
colvars:   # forceConstant = 1 [default]
colvars:   # decoupling = off [default]
colvars:   # targetForceConstant = -1 [default]
colvars:   The force constant for colvar "rmsd" will be rescaled to 10000 according to the specified width (0.01).
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new "harmonicwalls" instance.
colvars:   # name = "wall_rmsd"
colvars:   # colvars = { rmsd }
colvars:   # stepZeroData = off [default]
colvars:   # outputEnergy = off [default]
colvars:   # outputFreq = 0 [default]
colvars:   # timeStepFactor = 1 [default]
colvars:   # writeTISamples = off [default]
colvars:   # writeTIPMF = off [default]
colvars:   # forceConstant = 1 [default]
colvars:   # decoupling = off [default]
colvars:   # targetForceConstant = -1 [default]
colvars:   # lowerWalls = { 0 } [default]
colvars:   Lower walls were not provided.
colvars:   # upperWalls = { 0.1 }
colvars:   # upperWallConstant = 10
colvars:   The upper wall force constant for colvar "rmsd" will be rescaled to 100000 according to the specified width (0.01).
colvars: ----------------------------------------------------------------------
colvars: Collective variables biases initialized, 2 in total.
colvars: ----------------------------------------------------------------------
colvars: Collective variables module (re)initialized.
colvars: ----------------------------------------------------------------------
colvars: Re-initialized atom group for variable "rmsd":0/0. 10 atoms: total mass = 120.11, total charge = 0.53.
colvars: ----------------------------------------------------------------------
colvars: Reading new configuration:
colvars: # units = "" [default]
colvars: # smp = on [default]
colvars: # colvarsTrajFrequency = 1 [default]
colvars: # colvarsRestartFrequency = 0 [default]
colvars: # scriptedColvarForces = off [default]
colvars: # scriptingAfterBiases = off [default]
colvars: # sourceTclFile = "" [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = "rmsd-all"
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "" [default]
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerToOrigin = off [default]
colvars:       # centerToReference = off [default]
colvars:       # rotateToReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 51 atoms requested: total mass = 674.397, total charge = -7.64.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/da.pdb"
colvars:     # refPositionsCol = "" [default]
colvars:     Enabling "centerToReference" and "rotateToReference", to minimize RMSD before calculating it as a variable: if this is not the desired behavior, disable them explicitly within the "atoms" block.
colvars:     This is a standard minimum RMSD, derivatives of the optimal rotation will not be computed as they cancel out in the gradients.
colvars:   All components initialized.
colvars:   # timeStepFactor = 1 [default]
colvars:   # width = 1 [default]
colvars:   # lowerBoundary = 0 [default]
colvars:   # upperBoundary = 1 [default]
colvars:   # hardLowerBoundary = on [default]
colvars:   # hardUpperBoundary = off [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputTotalForce = off [default]
colvars:   # outputAppliedForce = off [default]
colvars:   # subtractAppliedForce = off [default]
colvars:   # runAve = off [default]
colvars:   # corrFunc = off [default]
colvars: ----------------------------------------------------------------------
colvars: Collective variables initialized, 2 in total.
colvars: ----------------------------------------------------------------------
colvars: Collective variables biases initialized, 2 in total.
colvars: ----------------------------------------------------------------------
colvars: Collective variables module (re)initialized.
colvars: ----------------------------------------------------------------------
colvars: Re-initialized atom group for variable "rmsd":0/0. 10 atoms: total mass = 120.11, total charge = 0.53.
colvars: Re-initialized atom group for variable "rmsd-all":0/0. 51 atoms: total mass = 674.397, total charge = -7.64.
